src/graph/fixtures.ts

```diff
@@ -3,96 +3,152 @@ import {
+ */
+export const testItemFragment = (mockInput: {
+  itemId: string;
+  __typename: 'PendingItem' | 'Item';
+}): ItemFragment => {
   return {
-    itemId: `item-${id}`,
-    resolvedId: `resolved-${id}`,
-    wordCount: 100,
-    title: 'title',
-    timeToRead: 10,
-    resolvedUrl: 'https://test.com',
-    givenUrl: 'https://test.com',
-    excerpt: 'excerpt',
-    domain: 'test.com',
-    isArticle: true,
-    isIndex: true,
-    hasVideo: Videoness.HasVideos,
-    hasImage: Imageness.HasImages,
-    language: 'en',
-    ampUrl: 'https://test.com',
-    topImage: {
-      url: 'https://test.com/image.jpg',
-    },
+    ...mockItemFragment,
+    itemId: mockInput.itemId,
+    resolvedId: `resolved-${mockInput.itemId}`,
   };
 };
 
+type MockListItemObject = Omit<ListItemObject, 'item_id' | 'resolved_id'> & {
+  ids: string[];
+};
 /**
  * return REST v3 GET response for given Ids
  * //todo: need to include tags, images, videoes etc
  * //todo: map top level fields and sort_id
  * @param ids
  */
 export const testV3GetResponse = (
-  ids: string[],
-  time_added?: string,
-  time_updated?: string
+  mockInputs: MockListItemObject = {
+    ...seedDataRest,
+  }
 ): RestResponse => {
   const map: { [key: string]: ListItemObject } = {};
-  ids.forEach((id) => {
+
+  mockInputs.ids.forEach((id) => {
     map[id] = {
       item_id: id,
       resolved_id: `resolved-${id}`,
-      given_url: 'https://test.com',
-      given_title: 'title',
-      favorite: '1',
-      status: '0',
-      //can add more test injection as we work on tickets
-      time_added: time_added ?? '1677818995',
-      time_updated: time_updated ?? '1677818995',
-      time_read: '1677818995',
-      time_favorited: '1677818995',
-      resolved_title: 'title',
-      resolved_url: 'https://test.com',
-      title: 'title',
-      excerpt: 'excerpt',
-      is_article: '1',
-      is_index: '1',
-      has_video: '1',
-      has_image: '1',
-      word_count: '100',
-      lang: 'en',
-      time_to_read: 10,
-      amp_url: 'https://test.com',
-      top_image_url: 'https://test.com/image.jpg',
+      title: mockInputs.title,
+      given_title: mockInputs.given_title,
+      given_url: mockInputs.given_url,
+      favorite: mockInputs.favorite,
+      status: mockInputs.status,
+      time_added: mockInputs.time_added,
+      time_updated: mockInputs.time_updated,
+      time_read: mockInputs.time_read,
+      time_favorited: mockInputs.time_favorited,
+      resolved_title: mockInputs.resolved_title,
+      resolved_url: mockInputs.resolved_url,
+      excerpt: mockInputs.excerpt,
+      is_article: mockInputs.is_article,
+      is_index: mockInputs.is_index,
+      has_video: mockInputs.has_video,
+      has_image: mockInputs.has_image,
+      word_count: mockInputs.word_count,
+      lang: mockInputs.lang,
+      time_to_read: mockInputs.time_to_read,
+      amp_url: mockInputs.amp_url,
+      top_image_url: mockInputs.top_image_url,
```

Comment:
Seems like other than `item_id` and`resolved_id`, all the other object prop names match the `mockInputs` prop names. So if you wanted to clean this up a bit, we could just spread the `mockInputs` object after line 127 🤔

Reply:
i attempted the spread first.. it includes `ids` in `convertSavedItemsToRestResponse().list` response and fails the test, so explicitly assigned
https://github.com/Pocket/v3-proxy-api/pull/9#discussion_r1134381034